[SPARK-5752][SQL] Don't implicitly convert RDDs directly to DataFrames

rxin opened the pull request:
- The old implicit would convert RDDs directly to DataFrames, and that added too many methods. 
- toDataFrame -> toDF
- Dsl -> functions
- implicits moved into SQLContext.implicits
- addColumn -> withColumn
- renameColumn -> withColumnRenamed

Python changes:
- toDataFrame -> toDF
- Dsl -> functions package
- addColumn -> withColumn
- renameColumn -> withColumnRenamed
- add toDF functions to RDD on SQLContext init
- add flatMap to DataFrame

rxin committed:
[SPARK-5752][SQL] Don't implicitly convert RDDs directly to DataFrames

rxin committed:
[SPARK-5752][SQL] Don't implicitly convert RDDs directly to DataFrames

SparkQA commented:
[Test build #27332 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27332/consoleFull) for   PR 4556 at commit [`ab58d66`](https://github.com/apache/spark/commit/ab58d66c2238b03328a14711f2e86d7950290088).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27332 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27332/consoleFull) for   PR 4556 at commit [`ab58d66`](https://github.com/apache/spark/commit/ab58d66c2238b03328a14711f2e86d7950290088).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `trait DataFrame extends RDDApi[Row] with Serializable `

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27332/
Test PASSed.

SparkQA commented:
[Test build #27371 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27371/consoleFull) for   PR 4556 at commit [`f755d21`](https://github.com/apache/spark/commit/f755d2129a663f027cade984acfd419281bb4222).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27371 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27371/consoleFull) for   PR 4556 at commit [`f755d21`](https://github.com/apache/spark/commit/f755d2129a663f027cade984acfd419281bb4222).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27371/
Test PASSed.

rxin committed:
- toDataFrame -> toDF
- Dsl -> functions
- implicits moved into SQLContext.implicits
- addColumn -> withColumn
- renameColumn -> withColumnRenamed

rxin committed:
- toDataFrame -> toDF
- Dsl -> functions
- implicits moved into SQLContext.implicits
- addColumn -> withColumn
- renameColumn -> withColumnRenamed

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27391/
Test FAILed.

rxin commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27394/
Test FAILed.

Commit:
Dsl -> functions, toDF()

Commit:
Dsl -> functions, toDF()

Commit:
schema() -> schema

Commit:
schema() -> schema

Commit:
add comment for toDF

Commit:
add comment for toDF

SparkQA commented:
[Test build #27397 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27397/consoleFull) for   PR 4556 at commit [`81e7685`](https://github.com/apache/spark/commit/81e7685d01faa109755fdfd9ff1371f35f568197).
 * This patch merges cleanly.

Commit:
add comment

Commit:
add comment

SparkQA commented:
[Test build #27397 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27397/consoleFull) for   PR 4556 at commit [`81e7685`](https://github.com/apache/spark/commit/81e7685d01faa109755fdfd9ff1371f35f568197).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27397/
Test FAILed.

SparkQA commented:
[Test build #27400 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27400/consoleFull) for   PR 4556 at commit [`ef8c510`](https://github.com/apache/spark/commit/ef8c510800a21ff5e29b0ba2fc4aec0e13f3e845).
 * This patch merges cleanly.

rxin committed:
style

rxin committed:
style

SparkQA commented:
[Test build #27400 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27400/consoleFull) for   PR 4556 at commit [`ef8c510`](https://github.com/apache/spark/commit/ef8c510800a21ff5e29b0ba2fc4aec0e13f3e845).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27400/
Test FAILed.

SparkQA commented:
[Test build #27401 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27401/consoleFull) for   PR 4556 at commit [`9e7f89e`](https://github.com/apache/spark/commit/9e7f89e460b0ed3e8b72acf7eb80f4a2282478dd).
 * This patch merges cleanly.

Commit:
flatMap

Commit:
flatMap

SparkQA commented:
[Test build #27402 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27402/consoleFull) for   PR 4556 at commit [`3858ad9`](https://github.com/apache/spark/commit/3858ad98ceb84f179ba917290fe27ba6f4338cd3).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27401 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27401/consoleFull) for   PR 4556 at commit [`9e7f89e`](https://github.com/apache/spark/commit/9e7f89e460b0ed3e8b72acf7eb80f4a2282478dd).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27401/
Test FAILed.

SparkQA commented:
[Test build #27402 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27402/consoleFull) for   PR 4556 at commit [`3858ad9`](https://github.com/apache/spark/commit/3858ad98ceb84f179ba917290fe27ba6f4338cd3).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27402/
Test FAILed.

rxin committed:
Fixed explode compilation break.

rxin committed:
Fixed explode compilation break.

SparkQA commented:
[Test build #27407 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27407/consoleFull) for   PR 4556 at commit [`b401335`](https://github.com/apache/spark/commit/b4013355988ede082e2fe51eebe6668e797377c2).
 * This patch merges cleanly.

rxin committed:
Fix build break again.

rxin committed:
Fix build break again.

SparkQA commented:
[Test build #27409 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27409/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27407 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27407/consoleFull) for   PR 4556 at commit [`b401335`](https://github.com/apache/spark/commit/b4013355988ede082e2fe51eebe6668e797377c2).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27407/
Test FAILed.

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27413 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27413/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27409 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27409/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27409/
Test FAILed.

SparkQA commented:
[Test build #27413 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27413/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27413/
Test FAILed.

rxin commented:
cc @davies Python test is failing

SparkQA commented:
[Test build #602 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/602/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch merges cleanly.

rxin committed:
count(*) fixes.

rxin committed:
count(*) fixes.

SparkQA commented:
[Test build #27421 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27421/consoleFull) for   PR 4556 at commit [`9e214d5`](https://github.com/apache/spark/commit/9e214d5b8b18f97814bcda2544f1d59c659528e8).
 * This patch merges cleanly.

Commit:
add test for agg('*')

Commit:
clean up, toPandas

SparkQA commented:
[Test build #602 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/602/consoleFull) for   PR 4556 at commit [`1ed7136`](https://github.com/apache/spark/commit/1ed7136f912e1ca813a949d1b689cbb16ee6d29c).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

mengxr commented on mllib/src/main/scala/org/apache/spark/ml/Transformer.scala:
```diff
@@ -97,7 +97,7 @@ private[ml] abstract class UnaryTransformer[IN, OUT, T <: UnaryTransformer[IN, O
-    dataset.select($"*", callUDF(
-      this.createTransformFunc(map), outputDataType, dataset(map(inputCol))).as(map(outputCol)))
+    dataset.withColumn(map(outputCol),
+      callUDF(this.createTransformFunc(map), outputDataType, dataset(map(inputCol))))
```
minor: use `udf`

mengxr commented on sql/core/src/main/scala/org/apache/spark/sql/functions.scala:
```diff
@@ -28,17 +29,9 @@ import org.apache.spark.sql.types._
-    }
-  }
+// scalastyle:off
+object functions {
```
minor: `// scalastyle:ignore`

mengxr commented on sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTest.scala:
```diff
@@ -90,7 +90,7 @@ trait ParquetTest {
     import sqlContext.implicits._
     withTempPath { file =>
-      sparkContext.parallelize(data).saveAsParquetFile(file.getCanonicalPath)
+      sparkContext.parallelize(data).toDF().saveAsParquetFile(file.getCanonicalPath)
```
minor: `toDF()` -> `toDF`

mengxr commented:
LGTM.

Commit:
fix test

SparkQA commented:
[Test build #27429 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27429/consoleFull) for   PR 4556 at commit [`e1d76e8`](https://github.com/apache/spark/commit/e1d76e8d2879c96b50ee6b973b5f1c8db00209fa).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27421 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27421/consoleFull) for   PR 4556 at commit [`9e214d5`](https://github.com/apache/spark/commit/9e214d5b8b18f97814bcda2544f1d59c659528e8).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27421/
Test FAILed.

SparkQA commented:
[Test build #27429 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27429/consoleFull) for   PR 4556 at commit [`e1d76e8`](https://github.com/apache/spark/commit/e1d76e8d2879c96b50ee6b973b5f1c8db00209fa).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27429/
Test FAILed.

Commit:
fix test

Commit:
fix mistake

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27444 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27444/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27444 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27444/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27444/
Test FAILed.

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27451 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27451/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27451 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27451/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27451/
Test FAILed.

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27452 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27452/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch merges cleanly.

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27454 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27454/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27452 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27452/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27452/
Test FAILed.

SparkQA commented:
[Test build #27454 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27454/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27454/
Test FAILed.

rxin commented:
Jenkins, retest this please.

SparkQA commented:
[Test build #27459 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27459/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27459 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27459/consoleFull) for   PR 4556 at commit [`a4aaf58`](https://github.com/apache/spark/commit/a4aaf58a7c6183fcfba1ad3a41d75220dd5ea220).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27459/
Test FAILed.

rxin committed:
Fix python

SparkQA commented:
[Test build #27465 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27465/consoleFull) for   PR 4556 at commit [`ff5832c`](https://github.com/apache/spark/commit/ff5832c1200427d436475942f3f23cff5700aa53).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27465 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27465/consoleFull) for   PR 4556 at commit [`ff5832c`](https://github.com/apache/spark/commit/ff5832c1200427d436475942f3f23cff5700aa53).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27465/
Test FAILed.

rxin committed:
Merge branch 'df5' of github.com:davies/spark into SPARK-5752

Conflicts:
	python/pyspark/sql/dataframe.py
	python/pyspark/sql/functions.py

SparkQA commented:
[Test build #27471 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27471/consoleFull) for   PR 4556 at commit [`61d3fca`](https://github.com/apache/spark/commit/61d3fca0eb66a964978c094b924a90c589eb370b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27471 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27471/consoleFull) for   PR 4556 at commit [`61d3fca`](https://github.com/apache/spark/commit/61d3fca0eb66a964978c094b924a90c589eb370b).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27471/
Test FAILed.

rxin committed:
More fix

SparkQA commented:
[Test build #27476 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27476/consoleFull) for   PR 4556 at commit [`5ef9910`](https://github.com/apache/spark/commit/5ef99109f8ab5d1aec086a27b7878dbe1998c1bd).
 * This patch merges cleanly.

SparkQA commented:
[Test build #27476 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/27476/consoleFull) for   PR 4556 at commit [`5ef9910`](https://github.com/apache/spark/commit/5ef99109f8ab5d1aec086a27b7878dbe1998c1bd).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/27476/
Test PASSed.
